nexus/types/src/deployment.rs

```diff
@@ -254,6 +254,18 @@ impl Blueprint {
     pub fn display(&self) -> BlueprintDisplay<'_> {
         BlueprintDisplay { blueprint: self }
     }
+
+    // ---
+    // Helper methods
+    // ---
+
+    fn diff_metadata(&self) -> BlueprintDiffMetadata {
```

Comment:
What exactly is "metadata" (or: why are the internal and external DNS not just "another part of the blueprint" like zones and future stuff like which host OS should be running on which sleds, etc)?

Reply:
So I needed an abstraction to represent that:

* the "before" of a diff can be either a collection or a blueprint
* the "after" is always a blueprint
* the blueprint has stuff associated with it that collections don't.

So `DiffBlueprintMetadata` consists of all the blueprint-specific things, and there's also a `DiffBeforeMetadata` that has `Collection` and `Blueprint` variants.

Reply:
Updated this to the `BlueprintMetadata` that was added later.